src/magma/vk/descriptors/descriptorSetLayout.cpp

```diff
@@ -162,6 +174,32 @@ void DescriptorSetLayout::bindCombinedImageSampler(uint32_t binding, VkImageView
     descriptorWrite.pImageInfo = imageInfo;
 }
 
+void DescriptorSetLayout::bindStorageImage(uint32_t binding, VkImageView imageView, VkSampler sampler, VkImageLayout imageLayout)
```

Comment:
Argument VkSampler sampler looks useless here.
Did you try pushing VK_NULL_HANDLE to imageInfo->sampler?